browser/todo-list.md

```diff
@@ -3,291 +3,248 @@
+  });
 };
 ```
 
-Once the user authenticates, the application handles the `authResponse` in the `src/components/Profile.js` file. :
+These todos are subsequently loaded using the [`getFile`](http://blockstack.github.io/blockstack.js/globals.html#getfile) method of the same object in the same module:
 
 ```js
-...
-componentWillMount() {
-    if (userSession.isSignInPending()) {
-      userSession.handlePendingSignIn().then((userData) => {
-        //if (!userData.username) {
-        //  throw new Error('This app requires a username.')
-        //}
-        window.location = window.location.origin;
-      });
-    }
-  }
-...
+export const fetchTasks = async (userSession, username) => {
+  const tasksJSON = await userSession.getFile(TASKS_FILENAME, {
+    decrypt: false,
+    username: username || undefined,
+  });
+  // code to format and return the tasks
+};
 ```
 
-If [`isUserSignedIn()`](https://blockstack.github.io/blockstack.js/#isusersignedin) is true, the user was previously signed in so Blockstack pulls the data from the browser and uses it in our application. If the check on  [`UserSession.isSignInPending()`](https://blockstack.github.io/blockstack.js/#issigninpending) is true, a previous `authResponse` was sent to the application but hasn't been processed yet. The `handlePendingSignIn()` function processes any pending sign in.
+By default, the `putFile` and `getFile` methods automatically encrypt data when saved and decrypt it when retrieved, using the user's Secret Key. This ensures that only the user has the ability to view this data.
 
-Signout is handled in `src/components/App.js`.
+When deleting a todo, the same `putFile` method is used to save a new JSON array of todos that excludes the deleted todo.
 
-```js
-  handleSignOut(e) {
-    e.preventDefault();
-    userSession.signUserOut(window.location.origin);
-  }
-```
+### Publish your todos publicly
 
-The method allows the application creator to decide where to redirect the user upon Sign Out:
+Select "Make public" to make your todos accessible to the public for sharing via URL.
 
+![](images/todos-public.svg)
 
-## Task 6: Work with the application
+This will call `saveTasks` with the `isPublic` parameter set to `true`, which is used to disable encryption when using `putFile`.
 
-Now, trying adding a few items to the todo list. For example, try making a list of applications you want to see built on top of Blockstack:
+The app will now show all of your todos to anyone who visits the URL displayed with your Blockstack username as a suffix.
 
-![](images/make-a-list.png)
+### Sign out and see your public tasks
 
-Each list is immediately stored in the Gaia Hub linked to your Blockstack ID.
-For more information about the Gaia hub, [see the overview in this documentation]({{ site.baseurl }}/storage/overview.html#). Now that you have seen the application in action, dig into how it works.
+Select "Sign out" to deauthenticate the app with your Blockstack account.
```

Comment:
I feel like this could be clearer. Something like "logout your Blockstack account from the app" or "close the user session" or similar?